app/models/due_date.rb

```diff
@@ -54,5 +50,14 @@ def self.copy(due_dates, new_assignment_id)
     end
   end
 
+  # factory method for retrieving due dates based on type of the parent
+  def self.get_next_due_date
+    case self.parent_type
+    when 'AssignmentDueDate'
+      self.next_due_date(AssignmentDueDate.get_due_dates(self.parent_id))
+    when 'ProjectTopicDueDate'
+      self.next_due_date(TopicDueDate.get_due_dates(self.parent_id.assignment, self.parent_id))
+    end
+  end
```

Comment:
Best guess at what Dr Gehringer was getting at in our last meeting. Feel free to tweak or remove as you see fit :)